packages/astro/src/@types/astro.ts

```diff
@@ -1331,9 +1331,15 @@ export interface AstroConfig extends z.output<typeof AstroConfigSchema> {
 	// TypeScript still confirms zod validation matches this type.
 	integrations: AstroIntegration[];
 }
-export interface AstroInlineConfig extends AstroUserConfig {
+export interface AstroInlineConfig extends AstroUserConfig, AstroInlineOnlyConfig {}
+export interface AstroInlineOnlyConfig {
 	configFile?: string | false;
 	mode?: RuntimeMode;
+	logLevel?: LoggerLevel;
+	/**
+	 * @internal for testing only
+	 */
+	logging?: LogOptions;
```

Comment:
Adds two new options to the inline config. I'm not really happy to add `logging` here, but if we can get Ema's `Logger` proposal, maybe we can substitute to that instead which is more user-friendly.

Reply:
I totally agree